lib/utils/getContract/index.ts

```diff
@@ -0,0 +1,63 @@
+import { ethers, providers, Signer } from "ethers";
+
+import { addresses } from "../../constants";
+import IERC20 from "../../abi/IERC20.json";
+import WSKLIMA from "../../abi/wsKlima.json";
+import DistributorContractv4 from "../../abi/DistributorContractv4.json";
+import PunkTLD from "../../abi/PunkTLD.json";
+import PairContract from "../../abi/PairContract.json";
+import KlimaRetirementAggregator from "../../abi/KlimaRetirementAggregator.json";
+import ExercisePKlima from "../../abi/ExercisepKLIMA.json";
+import KlimaStakingHelper from "../../abi/KlimaStakingHelper.json";
+import KlimaStakingv2 from "../../abi/KlimaStakingv2.json";
+import KlimaRetirementStorage from "../../abi/KlimaRetirementStorage.json";
```

Comment:
see my discord dm about tree-shaking...

we **may** need to rethink this to optimize that certain views like `/pledge/` only load the relevant ABIs as all combined are ~150kb raw.

150kb is not the end of the world, though, and is probably minified further during the build/webpack, and nextjs initial pageload should still be fast because its server rendered html.

Soooo what do you think? Worth the effort to optimize?

Reply:
For the sake of simplicity I would leave the ABIs as is for now.
One contract, one Abi

Let's rather improve step-by-step